src/docs/markdown/caddyfile/common-uses.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Common Caddyfile Uses
+---
+
+# Common Caddyfile Uses
+
+This page presents some complete, minimal Caddyfile configurations for common use cases, which you might find helpful when building your own configs.
+
+These are not drop-in solutions; you will have to customize your domain name, ports/sockets, directory paths, etc. They are intended to illustrate only the basic requirements for the most common use cases.
+
+#### Menu
+
+- [Static file server](#static-file-server)
+- [Reverse proxy](#reverse-proxy)
+- [PHP](#php)
+- [Redirect `www.` subdomain](#redirect-www-subdomain)
+- [Trailing slashes](#trailing-slashes)
+
+
+## Static file server
+
+```
+example.com
+
+root * /var/www  # optional; default root is current directory
+file_server
+```
+
+
+## Reverse proxy
+
+All requests:
+
+```
+example.com
+
+reverse_proxy localhost:5000
+```
+
+Just requests having a path starting with `/api/`; static files for everything else:
+
+```
+example.com
+
+reverse_proxy /api/* localhost:5000
+root * /var/www  # optional; default root is current directory
+file_server
+```
+
+
+## PHP
+
+With FastCGI running, something like this works for most modern PHP apps:
+
+```
+example.com
+
+php_fastcgi unix//run/php/php-fpm.sock
```

Comment:
Still need to serve files in the 95% case.

```suggestion
php_fastcgi unix//run/php/php-fpm.sock

root * /var/www  # optional; default root is current directory 
file_server
```

Reply:
derp the next one says that (but tbh they almost always will)

Reply:
Actually yeah - we definitely should always note to add file_server here. I can't think of really any situations where it would _only_ be PHP. I guess if it's a JSON API or something... but...

Reply:
I think it's important for the user to know that they are separate things, we definitely don't want them serving static files if they shouldn't be...

Reply:
Okay then we add a comment `# the below will serve static files` or something. I'd rather avoid people jumping to the forums saying "I'm getting empty 200s", cause we already get plenty of those!

Reply:
I agree, the first example should be the most common pattern (include the `file_server` directive).  In fact I would just go with one example, and have text after it that points out that the `file_server` directive isn't necessary if the user knows they don't need to serve static files.